src/visuallyCompleteCalculator.ts

```diff
@@ -33,26 +30,65 @@ export type Metric = {
   };
 };
 
-export type MetricSubscriber = (measurement: Metric) => void;
+export type Metric = CommonMetadata & {
+  // value of "TTVC"
+  duration: number;
+};
+
+// currently, the only source of error is cancellation
+export type CancellationError = CommonMetadata & {
+  // reason for cancellation
+  cancellationReason: CancellationReason;
+
+  // Optional type of event that triggered cancellation
+  eventType?: string;
+
+  // Optional target of event that triggered cancellation
+  eventTarget?: EventTarget;
+};
```

Comment:
Hmm, what do you think about trying to make the CancellationError type also conform to the [`PerformanceMeasure`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceMeasure) specification?

I think the main change would be ensuring that all metadata (including cancellationReason goes into the detail bucket.

Nit: I would also just as soon not try to share a `CommonMetadata` interface for these two results.  Personally, I think it's a little more awkward to read and doesn't save that many characters.  😅

Reply:
I don't see a reason to align that to `PerformanceMeasure`, but if you feel strongly about it, I can move the data into `detail` to make it conform.

As for `CommonMetadata` - I actually find it easier to read that way. When I scan the code, I can quickly see `CommonMetadata` shared between the types and it saves me the mental cycles on thinking whether the two are actually alike or not.

Reply:
I put some more thought into it, and I think it might be awkward to try to conform to `PerformanceMeasure` for something that is not a _measure_ of time, but a _record_ of occurrence. I would like to keep it the way it is - or have a deeper conversation about where we see value in having error state conform to `PerformanceMeasure`.

Reply:
I'm pretty okay with your rationale for the cancellation type diverging from `PerformanceMeasure` schema.  👍 

However, I think that makes me want to push a little further on the awkwardness of `CommonMetadata`.  I would like it to be clearer that `Metric` is a data type that is designed to look similar to `PerformanceMeasure` and `CancellationError` is not.

Maybe we could change the structure so that _no_ fields are nested in `detail` in `CancellationError`?  Another small bonus of that change would be that `didNetworkTimeOut?: boolean;` can be explicit again; it's _always_ present in a `Metric` and only _sometimes_ present in a `CancellationError`.

Reply:
Of course - that sounds good to me! I refactored the types to be distinctly separate, and added a note on `Metric`'s alignment with `PerformanceMeasure` for future reference. Thank you Andrew!